### PR TITLE
Select country logic

### PR DESCRIPTION
**Pull request by rakshakhegde**
https://www.pivotaltracker.com/story/show/168980808

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/selectcountry/SelectCountryModel.kt [hunk 1]
```diff
@@ -1,16 +1,16 @@
 package org.simple.clinic.selectcountry
 
 data class SelectCountryModel(
-    val supportedCountries: List<Country>?
+    val countries: List<Country>?
 ) {
 
-  fun hasFetchedCountries() = supportedCountries != null
+  fun hasFetchedCountries() = countries != null
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/selectcountry/SelectCountryModel.kt]: Please specify return type explicitly.

File: app/src/test/java/org/simple/clinic/selectcountry/SelectCountryUpdateTest.kt [hunk 2]
```diff
@@ -13,23 +13,25 @@ class SelectCountryUpdateTest {
...
 
+  private val bangladesh = Country(
+      code = "BD",
+      endpoint = URI("https://bd.simple.org"),
+      displayName = "Bangladesh",
+      isdCode = "880"
+  )
+
+  private val countries = listOf(
+      Country(
```
Comment (ragunathjawahar) [inline app/src/test/java/org/simple/clinic/selectcountry/SelectCountryUpdateTest.kt]: Can we extract this as a country as well just like we did with Bangaladesh?

File: app/src/test/java/org/simple/clinic/selectcountry/SelectCountryUpdateTest.kt [hunk 3]
```diff
@@ -97,4 +97,18 @@ class SelectCountryUpdateTest {
...
+  @Test
+  fun `when retry is clicked, then fetch manifest`() {
+    val model = defaultModel
+        .manifestFetchError(NetworkError)
+
+    spec
+        .given(model)
+        .whenEvent(RetryClicked)
+        .then(assertThatNext(
+            hasModel(SelectCountryModel.FETCHING),
```
Comment (ragunathjawahar) [inline app/src/test/java/org/simple/clinic/selectcountry/SelectCountryUpdateTest.kt]: This has to be a transition. (i.e.,) `SelectCountryModel.fetching()`. If it isn't, then we'll lose any information that was gathered between transitions.

Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/selectcountry/SelectCountryUpdateTest.kt]: Right, I'll make this change.

File: app/src/test/java/org/simple/clinic/selectcountry/SelectCountryUpdateTest.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/selectcountry/SelectCountryUpdateTest.kt]: Done.

File: app/src/test/java/org/simple/clinic/selectcountry/SelectCountryUpdateTest.kt (hunk 3 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/selectcountry/SelectCountryUpdateTest.kt]: Done.

**Pull request merged**